pkg/registry/util_provider.go

```diff
@@ -21,12 +21,8 @@ func (d DefaultUtilProvider) Relocator(writer io.Writer, tlsCfg TLSConfig, chang
 	}
 }
 
-func (d DefaultUtilProvider) SourceUploader(changeState bool) SourceUploader {
-	if changeState {
-		return DefaultSourceUploader{}
-	} else {
-		return DiscardSourceUploader{}
-	}
+func (d DefaultUtilProvider) SourceUploader(writer io.Writer, tlsCfg TLSConfig, changeState bool) SourceUploader {
+	return &DefaultSourceUploader{Relocator: d.Relocator(writer, tlsCfg, changeState)}
```

Comment:
This actually means we don't need a DiscardSourceUploader

Reply:
this is clean